lib/Parser/Parse.cpp

```diff
@@ -8837,8 +8845,8 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
             {
                 if (IsStrictMode())
                 {
-                    if ((buildAST && pnode->sxUni.pnode1->nop == knopName) ||
-                        (!buildAST && operandToken.tk == tkID))
+                    if ((buildAST && pnode->sxUni.pnode1->nop == knopName && !pnode->sxUni.pnode1->IsSpecialName()) ||
```

Comment:
>IsSpecialName [](start = 99, length = 13)

there is IsSpecialName variable also, does it have same meaning?

Reply:
There is a `isSpecialName` member of `ParseNode` which has the same meaning, this getter gets it. I don't think there's a local variable with that name in `Parser::ParseExpr` - there is one in `Parser::ParseTerm` but it has a slightly different meaning.

Reply:
Refactored this check into a helper to make it more clear.